src/client/Client.cc

```diff
@@ -10512,8 +10512,10 @@ int Client::lazyio_synchronize(int fd, loff_t offset, size_t count)
   Inode *in = f->inode.get();
   
   _fsync(f, true);
-  if (_release(in))
+  if (_release(in)) {
+    _getattr(in, CEPH_STAT_CAP_SIZE, f->actor_perms);
```

Comment:
Hmm...I approved this PR, but now I'm wondering if we should check for and return the error code from _getattr.

Also, should we instead do something like this order instead?
```
set CEPH_STAT_CAP_SIZE in wanted caps for this inode
check_caps (and maybe we should set NODELAY and SYNCHRONOUS flags here)
_getattr(...)
```
In principle, that should mean that the __getattr wouldn't require a separate round trip to the MDS in most cases.

Reply:
All tests ( https://github.com/ceph/ceph/pull/28834 ) pass just using _getattr() ( no check_caps() ). I am trying to think of cases where you would need to revoke the implemented caps and flush/release them ( i.e check_caps() ) in the context of lazyio_synchronize() . Nothing springs to my mind immediately. So is that check_caps() in lazyio_synchronize() necessary? @jtlayton @ukernel @batrick